src/main.c

```diff
@@ -279,15 +405,58 @@ void WINAPI EventCallback(PEVENT_RECORD ev)
     // This logic is here to support packet captures from older systems.
 
     if (!!(ev->EventHeader.EventDescriptor.Keyword & KW_PACKET_END)) {
-        PcapNgWriteEnhancedPacket(
-            OutFile,
-            AuxFragBuf,
-            AuxFragBufOffset + FragLength,
-            Iface->PcapNgIfIndex,
-            !!(ev->EventHeader.EventDescriptor.Keyword & KW_SEND),
-            TimeStamp.HighPart,
-            TimeStamp.LowPart,
-            ev->EventHeader.ProcessId);
+
+        if (ev->EventHeader.EventDescriptor.Keyword & KW_MEDIA_NATIVE_802_11 &&
+            AuxFragBuf[1] & 0x40) {
+            // Clear Protected bit in the case of 802.11
+            // Ndis captures will be decrypted in the etl file
+
+            AuxFragBuf[1] = AuxFragBuf[1] & 0xBF; // _1011_1111_ - Clear "Protected Flag"
+        }
+
+        if (AddMetadata) {
+            CombineMetadataWithPacket(
```

Comment:
Can this code be written without having two different codepaths that call PcapNgWriteEnhancedPacket? The current proposed change sort of hides the PcapNgWriteEnhancedPacket call inside a function (CombineMetadataWithPacket) which I wouldn't expect from its name to call it.

Also, I'd do PID= first in the comment independent of whether the wifi metadata is in the comment. It seems more consistent.

Reply:
Sure, this happened after trying to merge my code with the previous PR (which surprised me). But yes, I can do that.

Reply:
The updated code looks great, thanks.